server/interfaces/interfaces.go

```diff
@@ -225,6 +225,46 @@ type ExecutionService interface {
 	GetExecution(ctx context.Context, req *espb.GetExecutionRequest) (*espb.GetExecutionResponse, error)
 }
 
+// CommandRunner executes commands. Implementations should run untrusted commands
+// in sandboxed environments.
+type CommandRunner interface {
+	// Run the given command.
+	Run(ctx context.Context, command *repb.Command, workingDir string) *CommandResult
+}
+
+// CommandResult captures the output and details of an executed command.
+type CommandResult struct {
```

Comment:
Not sure whether there's any benefit of making this an interface or not; guessing not since most of the contained values are mutable anyway?